src/legacy/ui/public/agg_types/controls/time_interval.tsx

```diff
@@ -0,0 +1,171 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get, find } from 'lodash';
+import React, { useEffect } from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiComboBox, EuiComboBoxOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+import { AggParamOption } from '../agg_param';
+import { parseInterval } from '../utils';
+
+interface ComboBoxOption extends EuiComboBoxOptionProps {
+  key: string;
+}
+
+function TimeIntervalParamEditor({
+  agg,
+  aggParam,
+  editorConfig,
+  value,
+  setValue,
+  showValidation,
+  setValidity,
+}: AggParamEditorProps<string>) {
+  const timeBase: string = get(editorConfig, 'interval.timeBase');
+  const options = timeBase
+    ? []
+    : (aggParam.options || []).reduce(
+        (filtered: ComboBoxOption[], option: AggParamOption) => {
+          if (option.enabled ? option.enabled(agg) : true) {
+            filtered.push({ label: option.display, key: option.val });
+          }
+          return filtered;
+        },
+        [] as ComboBoxOption[]
+      );
+
+  let selectedOptions: ComboBoxOption[] = [];
+  let definedOption: ComboBoxOption | undefined;
+  if (value) {
+    definedOption = find(options, { key: value });
+    selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
+  }
+
+  const isValid = value ? (definedOption ? true : parseInterval(value, timeBase)) : false;
+  const interval = get(agg, 'buckets.getInterval') && agg.buckets.getInterval();
+  const scaledHepText =
+    interval && interval.scaled && isValid ? (
+      <strong className="eui-displayBlock">
+        <FormattedMessage
+          id="common.ui.aggTypes.timeInterval.scaledHelpText"
+          defaultMessage="Currently scaled to {bucketDescription}"
+          values={{ bucketDescription: get(interval, 'description') || '' }}
+        />{' '}
+        <EuiIconTip
+          position="right"
+          type="questionInCircle"
+          content={interval.scale <= 1 ? tooManyBucketsTooltip : tooLargeBucketsTooltip}
+        />
+      </strong>
+    ) : null;
+
+  const helpText = (
+    <>
+      {scaledHepText}
+      {get(editorConfig, 'interval.help') || selectOptionHelpText}
+    </>
+  );
+
+  const errors = [];
+
+  if (!isValid && value) {
+    errors.push(
+      i18n.translate('common.ui.aggTypes.timeInterval.invalidFormatErrorMessage', {
+        defaultMessage: 'Invalid interval format.',
+      })
+    );
+  }
+
+  const onCustomInterval = (customValue: string) => {
+    const normalizedCustomValue = customValue.trim();
+    setValue(normalizedCustomValue);
+
+    if (normalizedCustomValue ? parseInterval(normalizedCustomValue, timeBase) : false) {
```

Comment:
Could this be simplified as `if (normalizedCustomValue && isValidInterval(normalizedCustomValue, timeBase)) {`?